raco/algebra.py

```diff
@@ -366,7 +381,29 @@ def shortStr(self):
 """Logical Relational Algebra"""
 
 
-class Union(BinaryOperator):
+class IdenticalSchemeBinaryOperator(BinaryOperator):
+
+    """BinaryOperator where both sides have the same schema"""
+
+    def partitioning(self):
+        """attributes that are partitioned in both inputs"""
+        return RepresentationProperties(hash_partitioned=set.intersection(
```

Comment:
I don't think the semantics assumed here are correct. Take two relations with identical schema `R1(x, y, z)`, `R2(x, y, z)`, where `R1` is partitioned on `(x, y)` and `R2` is partitioned on `(y, z)`.  This code will make the operator `union(R1, R2)` report that it is partitioned on the attribute `y`, which implies that all tuples in `R1` and `R2` with common values of `y` are present in the same partitions. But that is clearly not the case: a tuple hashed on `(x, y)` will not hash to the same value as a tuple hashed on `(y, z)`. Instead, we can assume identical partitioning if and only if the sets of partitioned attributes in the inputs are identical.